lib/elixir/lib/list.ex

```diff
@@ -591,7 +591,12 @@ defmodule List do
   """
   @spec to_string(:unicode.charlist) :: String.t
   def to_string(list) when is_list(list) do
-    case :unicode.characters_to_binary(list) do
+    try do
+       :unicode.characters_to_binary(list)
+    rescue
+      ArgumentError ->
+        raise ArgumentError, "cannot make this list into a string (use 'inspect(list)' for its internal representation as a string)"
```

Comment:
There are dozens of messages which structured like this: `"expected a char list, got: #{inspect(list)}"`, maybe it ok not mentioning `inspect/1`.